packages/application-config/src/types.ts

```diff
@@ -52,3 +52,86 @@ export type LoadingConfigOptions = {
   processEnv: NodeJS.ProcessEnv;
   applicationPath: string;
 };
+
+// Utility types
+type WordSeparators = '_' | '-';
+
+/**
+Represents an array of strings split using a given character or character set.
+Underscore characters are omitted.
+
+source: https://github.com/sindresorhus/type-fest/blob/fedbc441a314c1f9f5f6225c993860d0886261da/source/split.d.ts#L22:L29
+
+@example
+```
+declare function split<S extends string, D extends string>(string: S, separator: D): Split<S, D>;
+type Item = 'foo' | 'bar' | 'baz' | 'waldo';
+const items = 'foo,bar,baz,waldo';
+let array: Item[];
+array = split(items, ',');
+```
+*/
+type Split<
+  S extends string,
+  Delimiter extends string
+> = S extends `${infer Head}${Delimiter}${infer Tail}`
+  ? [Delimiter extends '_' ? `${Head}_` : Head, ...Split<Tail, Delimiter>]
+  : S extends Delimiter
+  ? []
+  : [S];
+
+/**
+Step by step takes the first item in an array literal, formats it and adds it to a string literal, and then recursively appends the remainder.
+Only to be used by `CamelCaseStringArray<>`.
+
+source: https://github.com/sindresorhus/type-fest/blob/fedbc441a314c1f9f5f6225c993860d0886261da/source/camel-case.d.ts#L11:L18
+*/
+type InnerCamelCaseStringArray<
+  Parts extends readonly unknown[],
+  PreviousPart
+> = Parts extends [`${infer FirstPart}`, ...infer RemainingParts]
+  ? FirstPart extends undefined
+    ? ''
+    : FirstPart extends ''
+    ? InnerCamelCaseStringArray<RemainingParts, PreviousPart>
+    : `${PreviousPart extends ''
+        ? FirstPart
+        : Capitalize<FirstPart>}${InnerCamelCaseStringArray<
+        RemainingParts,
+        FirstPart
+      >}`
+  : '';
+
+/**
+Starts fusing the output of `Split<>`, an array literal of strings, into a camel-cased string literal.
+It's separate from `InnerCamelCaseStringArray<>` to keep a clean API outwards to the rest of the code.
+
+source: https://github.com/sindresorhus/type-fest/blob/fedbc441a314c1f9f5f6225c993860d0886261da/source/camel-case.d.ts#L27:L30
+*/
+type CamelCaseStringArray<Parts extends readonly string[]> = Parts extends [
+  `${infer FirstPart}`,
+  ...infer RemainingParts
+]
+  ? Uncapitalize<`${FirstPart}${InnerCamelCaseStringArray<
+      RemainingParts,
+      FirstPart
+    >}`>
+  : never;
+
+/**
+Convert a string literal to camel-case.
+This can be useful when, for example, converting some kebab-cased command-line flags or a snake-cased database result.
+
+source: https://github.com/sindresorhus/type-fest/blob/fedbc441a314c1f9f5f6225c993860d0886261da/source/camel-case.d.ts#L73
+
+@example
+```
+const someVariable: CamelCase<'foo-bar'> = 'fooBar';
+const anotherVariable: CamelCase<'foo_bar'> = 'foo_Bar';
```

Comment:
These would be types and not consts, right?
```suggestion
type someVariable = CamelCase<'foo-bar'> = 'fooBar';
type anotherVariable = CamelCase<'foo_bar'> = 'foo_Bar';
```

Reply:
Thanks, I think the example is ok. I other words it's:
```ts
type SomeVariable = CamelCase<'foo-bar'>;
const someVariable: SomeVariable = 'fooBar';

type AnotherVariable = CamelCase<'foo-bar'>;
const anotherVariable: AnotherVariable = 'foo_Bar';
```

Reply:
Ah, yes. Now I see.
Thanks.